### PR TITLE
[IOTDB-2967] New writing process of cluster

### PR DESCRIPTION
**Pull request by MrQuansy**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+package org.apache.iotdb.db.mpp.sql.statement.crud;
+
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import java.util.List;
+
+/** BatchInsert contains multiple sub insert. */
+public interface BatchInsert {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java:28]:
```suggestion
public interface BatchInsertNode {
```
Move this interface to plan/node/write

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package org.apache.iotdb.db.mpp.sql.statement.crud;
+
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import java.util.List;
+
+/** BatchInsert contains multiple sub insert. */
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java:27]: describe which type of InsertNode should implement this

Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java:27]: describe which type of InsertNode should implement this

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java (hunk 1 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/BatchInsert.java:28]: Move this interface to plan/node/write

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/analyze/SchemaValidator.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class SchemaValidator {
+
+  private static final ISchemaFetcher schemaFetcher = ClusterSchemaFetcher.getInstance();
+
+  public static SchemaTree validate(InsertNode insertNode) {
+
+    SchemaTree schemaTree;
+    if (insertNode instanceof BatchInsert) {
+      BatchInsert batchInsert = (BatchInsert) insertNode;
+      schemaTree =
```
Comment (yifuzhou) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/analyze/SchemaValidator.java:36]: We have DataNodeSchemaCache for insert operation. And schemaFetcher has already applied in it，you can use this cache's methed directly.

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/write/InsertRowsNode.java [hunk 4]
```diff
@@ -100,6 +104,16 @@ public List<PlanNode> getChildren() {
   @Override
   public void addChild(PlanNode child) {}
 
+  @Override
```
Comment (yifuzhou) [inline server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/write/InsertRowsNode.java:107]: According to MPP design, planNode is only for describe the plan infomation, should not contain operation function. You can use visitor mode to add some operation in the PlanVisitor.

**Review by yifuzhou — COMMENTED** [review]
PTAL

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**